### ISSUE TITLE
Document how to extend a Horizon backend

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/horizon-docs#62]
Copying this comment from @sunjay https://github.com/rethinkdb/horizon/issues/466 here:

**Comment by danielmewes** [comment]
@Tryneus once things get a bit calmer, could you describe the basics of how to extend the Horizon server to perform tasks like this?

**Comment by fasiha** [comment]
My use case: I have an internal REST API that I want to secure behind Horizon's auth system. That is, I'd like to write a request handler for `/some-internal-api` that checks to make sure the user is authorized (using the same permissions system as the rest of Horizon), then forwarding the request to an internal URL and returning its response to the user.

(In my case, the internal REST API is unsecured because its users are assumed to deal with authentication/authorization.)

**Comment by sylvainlap** [comment]
+1

**Comment by ivan-kleshnin** [comment]
I've got a couple of questions which are not covered in docs. Of course, it's possible to investigate it manually but time is limited and you guys probably appreciate feedbacks ;)

So Horizon DB:

1) What `hz_` prefix actually *means*? Is it to underline that `users` belongs to app while `hz_groups` belong to HZ inner stuff? I can imagine I will output `hz_groups` content in admin panel so I'd like to hear "official" explanation.

2) Permissions are declared in`schema.toml` and copied to `groups` table? Couldn't we just declare them right in the table? Is it just for user convenience?

**Comment by danielmewes** [comment]
@ivan-kleshnin Yeah, essentially the `hz_` prefix means that this table contains metadata internal to Horizon. You can of course access it from your code if you want to, but its format can change in a future Horizon release. The contents of those tables are not part of the official Horizon API.

The answer for 2) is similar. Partially, the `hz schema apply` command is there for convenience. Additionally, we only try to maintain compatibility for the `schema.toml` file format, but we might change the *internal* format in the `hz_groups` table between versions.
If you don't mind that, you can write to the table directly as well.
(note that `hz schema apply` does more than just adding permissions. It also registers and creates collections and secondary indexes)

**Comment by ivan-kleshnin** [comment]
@danielmewes thank you, that makes sense.

**Comment by timothyclemansinsea** [comment]
@danielmewes I need to implement https://stripe.com/docs/charges and have no clue how to add my own backend code
